src/main/java/wooteco/subway/admin/service/LineService.java

```diff
@@ -25,30 +31,76 @@ public Line save(Line line) {
         return lineRepository.save(line);
     }
 
-    public List<Line> showLines() {
-        return lineRepository.findAll();
+    public List<LineResponse> showLines() {
+        List<LineResponse> lineResponses = new ArrayList<>();
+        final List<Line> lines = lineRepository.findAll();
+        for (Line line : lines) {
+            lineResponses.add(
+                LineResponse.convert(line, findStationsByStationIds(line.findLineStationsId())));
+        }
+        return lineResponses;
     }
 
-    public void updateLine(Long id, Line line) {
-        Line persistLine = lineRepository.findById(id).orElseThrow(RuntimeException::new);
+    public void updateLine(Long lineId, Line line) {
+        Line persistLine = findLineByLineId(lineId);
         persistLine.update(line);
         lineRepository.save(persistLine);
     }
 
-    public void deleteLineById(Long id) {
-        lineRepository.deleteById(id);
+    private Line findLineByLineId(Long lineId) {
+        return lineRepository.findById(lineId).orElseThrow(IllegalArgumentException::new);
+    }
+
+    public void deleteLineById(Long lineId) {
+        lineRepository.deleteById(lineId);
     }
 
     public void addLineStation(Long id, LineStationCreateRequest request) {
-        // TODO: 구현
+        Line line = findLineByLineId(id);
+        line.addLineStation(request.toLineStation());
+        lineRepository.save(line);
     }
 
     public void removeLineStation(Long lineId, Long stationId) {
-        // TODO: 구현
+        Line line = findLineByLineId(lineId);
+        line.removeLineStationById(stationId);
+        lineRepository.save(line);
+    }
+
+    public LineResponse findLineWithStationsByLineId(Long lineId) {
+        final Line line = lineRepository.findById(lineId)
+            .orElseThrow(() -> new IllegalArgumentException("해당 아이디의 노선이 존재하지 않습니다."));
+        return LineResponse.convert(line,
+            findStationsByStationIds(line.findLineStationsId()));
+    }
+
+    public Set<Station> findStationsByStationIds(List<Long> stationIds) {
+        return Sets.newHashSet(stationRepository.findAllById(stationIds));
```

Comment:
```suggestion
        return stationRepository.findAllById(stationIds);
```
stationRepository에 findAll() 메서드처럼 findAllById를 override하면 set으로 새로 감쌀 필요 없지 않을까요?